src/main/java/org/testng/xml/XmlSuite.java

```diff
@@ -1017,5 +1019,114 @@ public void addTest(XmlTest test) {
     }
     return result;
   }
+
+  private XmlSuite cleanClone(List<XmlTest> tests) {
```

Comment:
I was just piggy backing on the fact that it already holds logic such as `shallowCopy()` and `clone()`

Reply:
XmlObjects should just be POJO for the xml binding without business logic, could you move these methods in an external util class?